rate_counter/rate/views.py

```diff
@@ -1,3 +1,161 @@
-from django.shortcuts import render
+from django.shortcuts import render, HttpResponseRedirect, get_object_or_404, get_list_or_404
+from .models import Institute, Subject, StudentGroup, Table, TableRow
+from .forms import TableForm
 
 # Create your views here.
+
+
+def create_table(request):
+    """
+    creation of new table
+    """
+    form = TableForm(request.POST or None)
+    if form.is_valid():
+        instance = form.save(commit=False)
+        instance.save()
+        return HttpResponseRedirect("/rate/detail/%s/" % instance.id)
+    context = {
+        'form': form
+    }
+    return render(request=request, template_name="table_form.html", context=context)
+
+
+def edit_table(request, id=None):
+    """
+    edit of existing table
+    """
+    table_obj = get_object_or_404(Table, id=id)
+    form = TableForm(request.POST or None, request.FILES or None, instance=table_obj)
+    if form.is_valid():
+        instance = form.save(commit=False)
+        instance.save()
+        return HttpResponseRedirect("/rate/detail/%s/" % instance.id)
+    context = {
+        'form': form
+    }
+    return render(request=request, template_name="table_form.html", context=context)
+
+
+def delete_table(id=None):
+    """
+    table removal
+    """
+    table_obj = Table.objects.get(id=id)
+    table_obj.delete()
+    return HttpResponseRedirect("/rate/")
+
+
+def tables_list(request):
+    """
+    get page with all tables
+    """
+    queryset = Table.objects.all()
+
+    context = {
+        "tables_list": queryset
+    }
+
+    return render(request, "tables_list.html", context)
+
+
+def table_detail(request, id=None):
+    """
+    get page with a particular table
+    """
+    inst_set = Institute.objects.all()
+    grp_set = StudentGroup.objects.all()
+    institute = request.POST.get('Institutes')
+    table_obj = get_object_or_404(Table, id=id)
+    table_rows = TableRow.objects.filter(table=table_obj)
+    coef_sum, point_sum, result = 0, 0, 0
+    if table_rows:
```

Comment:
I suggest to move counting of rate to separate function. This will give you a possibility to cover it with unit test, For example in some modules like 'utils'